docs/guides/server/logging.adoc

```diff
@@ -71,15 +74,90 @@ To enable log handlers, enter the following command:
 
-The available handlers are
-<@profile.ifCommunity>
-`console`, `file`, `syslog`.
-</@profile.ifCommunity>
-<@profile.ifProduct>
-`console`, `file` and `syslog`.
-</@profile.ifProduct>
+The available handlers are:
+
+<@opts.expectedValues option="log"/>
+
 The more specific handler configuration mentioned below will only take effect when the handler is added to this comma-separated list.
 
+== Specify log level for each handler
+
+The `log-level` property specifies the global root log level and levels for selected categories.
+However, a more fine-grained approach for log levels is necessary to comply with the modern application requirements.
+
+To set log levels for particular handlers, properties in format `log-<handler>-level` (where `<handler>` is available log handler) were introduced.
+
+It means properties for log level settings look like this:
+
+* `log-console-level` - Console log handler
+* `log-file-level` - File log handler
+* `log-syslog-level` - Syslog log handler
+
+NOTE: The `log-<handler>-level` properties are available only when the particular log handlers are enabled.
+More information in log handlers settings below.
+
+Accepted log levels are specified in <<Log levels>> section, and *must be in lowercase*.
+
+=== General principle
+
+It is necessary to understand that setting the log levels for each particular handler *does not override the root level* specified in the `log-level` property.
+It specifies the minimal allowed level for logs shown in the output.
+It respects levels specified in the `log-level` property, which creates a minimal log level bar for log records.
+
+Specifically, when an arbitrary log level is specified for the handler, it does not mean the log records with the log level will be present in the output.
+The root log level needs to be assessed as well, as it determines the minimal level for exposed log records.
+
+Log handler levels provide the *restriction for the root log level*, so the default value is `all` - without any restriction.
+
+==== Visualization
+
+To better understand the principle of logging levels, the visualization of log level settings might help.
+Only the `log-level` property is changed, and other log handlers are set as follows:
+
+<@kc.start parameters="--log=syslog,console,file --log-level=info --log-syslog-level=all --log-console-level=debug --log-file-level=trace"/>
+
+.Root log level `info`
+[#img-log-level-info]
+image::logging/log-levels-root-info.png[Root log level info,720]
+
+As you can see in the figure <<img-log-level-info>>, the root log level, specified via the `log-level` property, sets the minimal level for log records.
+Therefore, even when their log level is the finest, it is ignored and respects the parent level.
+
+The _Min Level_ bar represents the minimal allowed log level in the application (`trace` by default).
+Log records with finer log levels will never be present in {project_name}.
+It is controlled by the Quarkus property `quarkus.log.min-level`.
```

Comment:
I would not mention Quarkus options here. User cares about Keycloak and not care about implementational details.